lib/chef/data_collector/run_end_message.rb

```diff
@@ -60,8 +60,8 @@ def construct_message(data_collector, status)
             "cookbooks" => ( node && node["cookbooks"] ) || {},
             "policy_name" => node&.policy_name,
             "policy_group" => node&.policy_group,
-            "start_time" => run_status.start_time.utc.iso8601,
-            "end_time" => run_status.end_time.utc.iso8601,
+            "start_time" => run_status&.start_time&.utc&.iso8601,
+            "end_time" => run_status&.end_time&.utc&.iso8601,
```

Comment:
Added safe navigator operator to fix `FATAL: NoMethodError: undefined method `start_time' for nil:NilClass` accessing `start_time` on nil `run_context`.